common/lib/xmodule/xmodule/modulestore/mixed.py

```diff
@@ -172,8 +172,12 @@ def _get_modulestore_for_courseid(self, course_id=None):
             if mapping is not None:
                 return mapping
             else:
+                if isinstance(course_id, LibraryLocator):
```

Comment:
Sorry, that wasn't clear. My point is that instead of adding this logic into this method, I would move it into ``has_course`` so that the call below doesn't have to be changed. It doesn't make sense to me that only this method knows how to determine if a library is managed by a modulestore.

Part of the problem is that the parameter passed in is called ``course_id``, but it is no longer a course_id in the true meaning of the term. Is a library a course or not? If not, we need a term for a 'thing that is either a library or a course'. If a library is a course, then this logic unambiguously should move into store.has_course.